devel/create-github-release

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -euo pipefail
+shopt -s extglob
+
+devel="$(dirname "$0")"
+
+main() {
+    local version="${1:?version is required}"
+    local commit pre_release
+    shift
+    local -a assets=("$@")
+
+    if [[ ${#assets[@]} -eq 0 ]]; then
+        echo "ERROR: no assets provided" >&2
+        exit 1
+    fi
+
+    # Asserts that $version is an actual tag, not a branch name or other ref.
+    #
+    # Prefixing with refs/tags/ is a bit hacky and suffixing with ^{tag} would
+    # be more proper to restrict to a proper annotated, meant-for-release tag.
+    # While this is dandy in any given clone, in the GitHub Actions context the
+    # annotated tag is intentionally **overwritten** and converted to a
+    # lightweight tag by actions/checkout@v3, which doesn't/can't pass the
+    # muster of ^{tag}.
+    git rev-parse --verify "refs/tags/$version" >/dev/null
```

Comment:
_(non-blocking)_

You probably already saw this, but `git fetch --tags --force` from https://github.com/actions/checkout/issues/290#issuecomment-680260080 is another option:

```suggestion
    # Asserts that $version is an actual tag, not a branch name or other ref.
    #
    # In the GitHub Actions context the annotated tag is intentionally
    # **overwritten** and converted to a lightweight tag by
    # actions/checkout@v3. Overwrite the tags in original form so ^{tag}
    # (which only works on annotated tags) can be used properly.
    git fetch --tags --force
    git rev-parse --verify "$version^{tag}" >/dev/null
```

Reply:
Ah, I actually hadn't considered manually fetching the actual annotated tag object, but that's maybe a better idea. I would want to do that inside the workflow though, not inside this program, to properly scope the solution to the context in which it occurs (and ensure that if the program is run locally it doesn't muck around in your clone). I'd probably also not use the hammer of `git fetch --tags --force` but instead `git fetch --force origin "$GITHUB_REF"` or `git fetch --force origin tag "$GITHUB_REF_NAME"`.

Reply:
Implemented as 203adee.